### PR TITLE
fix: [CDS-50939]: application api payload fix

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger jest

File: src/modules/75-cd/pages/get-started-with-cd/Deploy/Deploy.tsx [hunk 1]
```diff
@@ -68,7 +68,7 @@ export const Deploy = ({ onBack, setSelectedSectionId }: DeployProps) => {
     const payload = getAppPayload({
       repositoryData,
       clusterData,
-      name: formikRef?.current?.values?.name
+      name: 'hostedapp'
```
Comment (rameshx) [inline src/modules/75-cd/pages/get-started-with-cd/Deploy/Deploy.tsx:71]: Will this always be a fixed value?

**Review by rameshx — DISMISSED** [review]

File: src/modules/75-cd/pages/get-started-with-cd/Deploy/Deploy.tsx (hunk 1 above)
Comment (ishantmehtaharness) [reply src/modules/75-cd/pages/get-started-with-cd/Deploy/Deploy.tsx:71]: yes

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx [hunk 2]
```diff
@@ -536,11 +538,30 @@ export interface PayloadInterface {
...
+}
+
+export enum ParameterType {
+  Directory = 'Directory',
+  Helm = 'Helm',
+  Kustomize = 'Kustomize',
+  Ksonnet = 'Ksonnet'
+}
+
+export interface VariableArray {
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:551]: Not needed

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx [hunk 3]
```diff
@@ -581,6 +606,18 @@ export const getAppPayload = (props: PayloadInterface) => {
       }
     }
   }
+
+  if (appDetails?.type === ParameterType.Directory) {
+    delete payload.application.spec.source.helm
+  }
+  if (appDetails?.type === ParameterType.Helm) {
+    delete payload.application.spec.source.directory
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:614]: here, we should delete kustomize also.
we should check with BE first

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx [hunk 4]
```diff
@@ -581,6 +606,18 @@ export const getAppPayload = (props: PayloadInterface) => {
       }
     }
   }
+
+  if (appDetails?.type === ParameterType.Directory) {
+    delete payload.application.spec.source.helm
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:611]: here, we should delete kustomize also.
we should check with BE first

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/ConfigureGitops.tsx [hunk 5]
```diff
@@ -453,8 +484,8 @@ const ConfigureGitopsRef = (props: any): JSX.Element => {
     } else {
       return [
         {
-          label: getString('cd.getStartedWithCD.master'),
-          value: getString('cd.getStartedWithCD.master')
+          label: getString('cd.getStartedWithCD.main'),
+          value: getString('cd.getStartedWithCD.main')
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/ConfigureGitops.tsx:488]: values should not be used from getString

File: src/modules/75-cd/pages/get-started-with-cd/Deploy/Deploy.tsx [hunk 6]
```diff
@@ -339,11 +350,20 @@ export const Deploy = ({ onBack, setSelectedSectionId }: DeployProps) => {
...
+              <Icon name="steps-spinner" color={Color.PRIMARY_7} />
+              <Text font={{ variation: FontVariation.BODY2 }} color={Color.PRIMARY_7}>
+                {getString('cd.getStartedWithCD.creatingApp')}
+              </Text>
+            </Layout.Horizontal>
+          ) : (
+            <Button
+              text={getString('cd.getStartedWithCD.syncApplication')}
+              variation={ButtonVariation.PRIMARY}
+              onClick={goToAppDetailPage}
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/Deploy/Deploy.tsx:364]: We can use loading prop

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger jest

**Review by ishantmehtaharness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**